the scanner is now less eager about deprecations

xeno-by opened the pull request:
When healing braces it isn't very useful to report deprecation warnings,
especially since this process is just simple context-free skimming, which
can't know about what positions can accept what identifiers.

xeno-by committed:
the scanner is now less eager about deprecations

When healing braces it isn't very useful to report deprecation warnings,
especially since this process is just simple context-free skimming, which
can't know about what positions can accept what identifiers.

xeno-by commented:
Supersedes https://github.com/scala/scala/pull/1763

xeno-by commented:
review @paulp 

btw, in a totally random fashion I stumbled upon a repro for the issue fixed in this patch :)

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/1227/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/1227/

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1945/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1945/

paulp commented on test/files/neg/macro-false-deprecation-warning.flags:
```diff
@@ -0,0 +1 @@
+-language:experimental.macros
```
The test doesn't test anything unless "-deprecation" is given. (I tried compiling the test case without it and it produces the checkfile, and doesn't even say "there were deprecation warnings.")

xeno-by replied:
Here's what I get when commenting out the `override def emitIdentifierDeprecationWarnings = false`:

```
22:19 ~/Projects/Kepler_ident-deprecation-warnings/test (topic/ident-deprecation-warnings)$ ./partest files/neg/macro-false-deprecation-warning

Testing individual files
testing: [...]/files/neg/macro-false-deprecation-warning              [FAILED]
1,3d0
< Impls_Macros_1.scala:13: warning: macro is now a reserved word; usage as an identifier is deprecated
<     def unapplySeq[T](x: List[T]): Option[Seq[T]] = macro impl[T]
<                                                     ^
7d3
< one warning found
1 of 1 tests failed (elapsed time: 00:00:01)
```

xeno-by replied:
Omg Paul I thought I was going mad, but apparently it's partest who silently enables `-deprecation` for all tests:

https://github.com/scalamacros/kepler/blob/e5d34d70499504e085ddf957c1c818ffb63f4e8d/src/partest/scala/tools/partest/nest/CompileManager.scala#L44

```
class TestSettings(cp: String, error: String => Unit) extends Settings(error) {
  def this(cp: String) = this(cp, _ => ())

  deprecation.value = true
  nowarnings.value  = false
  encoding.value    = "UTF-8"
  classpath.value   = cp
}
```

paulp replied:
Nice. I knew about that (not why, only that it exists) but forgot. #1824.

xeno-by committed:
changes the flags to not depend on partest

Due to some reason, partest always enables -deprecation. Since Paul has
just submitted a pull request, which removes this behavior, I'm updating
the flags to make sure this test works even after Paul's change.

xeno-by commented:
PLS REBUILD ALL

scala-jenkins commented:
Started jenkins job pr-scala-testsuite-linux-opt at https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1978/

scala-jenkins commented:
jenkins job pr-scala-testsuite-linux-opt: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-scala-testsuite-linux-opt/1978/

scala-jenkins commented:
Started jenkins job pr-rangepos at https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/1259/

scala-jenkins commented:
jenkins job pr-rangepos: Success - https://scala-webapps.epfl.ch/jenkins/job/pr-rangepos/1259/

paulp commented:
Still a little lukewarm, but lgtm.
